react/features/cropPerson/actions.js

```diff
@@ -0,0 +1,85 @@
+// @flow
+
+import { getLocalPresenterTrack,
+         getLocalVideoTrack
+} from '../../features/base/tracks';
+
+import { CROP_DISABLED, CROP_ENABLED } from './actionTypes';
+import { getCropPersonEffect } from './functions';
+import logger from './logger';
+import UIEvents from '../../../service/UI/UIEvents';
+
+declare var APP: Object;
+
+/**
+* Signals the local participant is switching between cropped or non cropped video.
+*
+* @param {boolean} enabled - If true enables video blur, false otherwise.
+* @returns {Promise}
+*/
+export function toggleCropPersonEffect(enabled: boolean) {
+    return function(dispatch: (Object) => Object, getState: () => any) {
+        const state = getState();
+        const presenterTrack 
+            = getLocalPresenterTrack(state['features/base/tracks']);
+        const videoTrack
+            = getLocalVideoTrack(state['features/base/tracks']);
+
+        // Apply crop effect only when presenter track and video
+        // track are present.
+        if (!presenterTrack || !videoTrack) {
+            logger.debug('Crop can be enabled only in presenter mode');
+
+            return Promise.resolve();
+        }
+
+        if (state['features/cropPerson'].cropEnabled === enabled) {
+
+            return Promise.resolve();
+        }
+
+        return getCropPersonEffect(presenterTrack.jitsiTrack.stream)
```

Comment:
Maybe try to write this with async / await so it looks a bit cleaner / easier to follow?